website/docs: Reorganize policy documentation (#21133)
goauthentik/authentik: website/docs/customize/policies/working_with_policies.md

---
title: Working with policies
tags:
    - policy
    - access-control
    - how-to
---

> Policies can be built-in policy types or expression policies.

Reviewer: ```suggestion
You can use built-in policies or create expression policies.
```

Policies can be built-in policy types or expression policies.

authentik provides several [standard policy types](./index.md#standard-policies), which can be configured for your specific needs. We also document several useful [expression policies](./expression.mdx#sample-expression-policies).

:::info

> You can add expressions to our standard policies to further customize them.

Reviewer 2: Let's not remove this... this is important and your rewrite above ("Policies can be built-in policy types or expression policies.") obfuscates how/where to do an expression...

> For an overview of the available policy types, see [Policies](./index.md). If you need background on how bindings behave after you attach them, see [Policy bindings and evaluation](./bindings.md).

Reviewer 2: This is an unnecessary rewrite, sir. The current way has two links, to very specific things.. let's not conflate them into a generic "see more" link.

1. Log in to authentik as an administrator and open the authentik Admin interface.
2. Navigate to **Customization** > **Policies**.
3. Click **Create**.

> 4. Select the policy type.

Reviewer 2: I wish the UI didn't say "policy type". Just select a policy... I'd suggest here you just say "Select the policy".. using the word type introduces a lot of cognitive effort.

5. Configure the policy-specific settings.
6. Click **Finish**.

> If you are not sure which policy type to choose, see [Types of policies in authentik](./types/index.mdx).

Reviewer 2: nice...

## Bind a policy to a flow, stage, application, or source

> After creating a policy, bind it to the place where you want the check to apply:

Reviewer 2: "the place" is quite vague. The current phrasing works fine...

4. Define the policy and click **Finish**.
3. Click **Create**.
4. Select the policy type.
5. Configure the policy-specific settings.
6. Click **Finish**.

If you are not sure which policy type to choose, see [Types of policies in authentik](./types/index.mdx).

## Bind a policy to a flow, stage, application, or source

After creating the policy, you can bind it to either a:
After creating a policy, bind it to the place where you want the check to apply:

- [flow](../../add-secure-apps/flows-stages/flow/index.md)
- [stage](../../add-secure-apps/flows-stages/stages/index.md)
- [stage binding](../../add-secure-apps/flows-stages/stages/index.md)
- [application](../../add-secure-apps/applications/index.md)
- [source](../../users-sources/sources/index.md)

:::info

> Bindings are instantiated objects themselves, and conceptually can be considered the "connector" between the policy and the component to which it is bound. This is why you might read about "binding a binding", because technically, a binding is "spliced" into another binding in order to intercept and enforce the criteria defined in the policy. To learn more, refer to our [Bindings documentation](../../add-secure-apps/bindings-overview/index.md).

Reviewer 2: I don't want to remove this. The concept of a stage binding is important.. as is a binding to abinding. Did you just move it somewhere else? Let's not just chop it out.

> Flow bindings control who can use the flow at all.

Reviewer 2: uneccessary rewrite. "Access" is the correct verb.

> 3. Open the flow to which you want to bind a policy.

Reviewer 2: leave original wording... we need to tell the user WHERE and HOW to "open" the flow (by clicking on the name).
